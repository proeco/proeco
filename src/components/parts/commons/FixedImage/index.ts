export { FixedImage, SkeltonFixedImage } from './FixedImage';
