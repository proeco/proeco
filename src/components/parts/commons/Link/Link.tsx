import React, { VFC } from 'react';
import NextLink, { LinkProps as NextLinkProps } from 'next/link';
import { Link as MuiLink } from '@mui/material';

type Props = {
  href: NextLinkProps['href'];
  target?: string;
  children?: React.ReactNode;
  underline?: 'none' | 'hover' | 'always';
};

export const Link: VFC<Props> = ({ href, target, children, underline = 'none' }) => (
  <NextLink href={href} passHref>
    <MuiLink target={target || '_self'} rel="noopener noreferrer" underline={underline}>
      {children}
    </MuiLink>
  </NextLink>
);
