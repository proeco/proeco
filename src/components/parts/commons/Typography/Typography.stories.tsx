import React from 'react';

import { Box } from '@mui/system';
import { Typography } from './Typography';

export default {
  title: 'parts/commons/Typography',
};

const Template = () => (
  <Box display="flex" flexDirection="column" gap="20px">
    <Typography variant="h1">h1: 48px</Typography>
    <Typography variant="h2">h2: 32px</Typography>
    <Typography variant="h3">h3: 24px</Typography>
    <Typography variant="h4">h4: 20px</Typography>
    <Typography variant="body1">body1: 16px</Typography>
    <Typography variant="body2">body2: 14px</Typography>
    <Typography variant="button">button: 14px</Typography>
    <Typography variant="caption">caption: 12px</Typography>
    <Typography variant="overline">overline: 10px</Typography>
  </Box>
);

export const Default = Template.bind({});
