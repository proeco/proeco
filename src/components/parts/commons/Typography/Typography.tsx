import React, { VFC, ComponentProps } from 'react';
import { Typography as MuiTypography } from '@mui/material';
import { styled } from '@mui/material/styles';

type Bold = {
  bold?: boolean;
  MaximumLines?: number;
};

type Props = ComponentProps<typeof MuiTypography> & Bold;

export const Typography: VFC<Props> = ({ bold, MaximumLines = 1, color = 'textColor.main', ...rest }) => {
  return <StyledTypography color={color} bold={bold ? 1 : 0} MaximumLines={MaximumLines} {...rest} />;
};

const StyledTypography = styled(MuiTypography)<{ bold: number; MaximumLines: number }>`
  text-transform: none;
  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
  display: -webkit-box;
  overflow: hidden;
  -webkit-line-clamp: ${(props) => props.MaximumLines};
  -webkit-box-orient: vertical;
`;
