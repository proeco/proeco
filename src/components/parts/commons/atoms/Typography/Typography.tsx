import React, { VFC, ComponentProps } from 'react';
import { Typography as MuiTypography } from '@mui/material';
import styled from 'styled-components';

type Bold = {
  bold?: boolean;
};

type Props = ComponentProps<typeof MuiTypography> & Bold;

export const Typography: VFC<Props> = ({ bold, ...rest }) => {
  return <StyledTypography bold={bold} {...rest} />;
};

const StyledTypography = styled(MuiTypography)<{ bold?: boolean }>`
  text-transform: none;
  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
`;
