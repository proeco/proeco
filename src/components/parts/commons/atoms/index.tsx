export { Button } from './Button';
export { Card } from './Card';
export { Divider } from './Divider';
export { TextField } from './TextField';
export { Typography } from './Typography';
