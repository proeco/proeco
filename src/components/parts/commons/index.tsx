export { Button } from './Button';
export { Card } from './Card';
export { CircularProgress } from './CircularProgress';
export { Divider } from './Divider';
export { EditableText } from './EditableText';
export { Editor } from './Editor';
export { Emoji } from './Emoji';
export { EmojiRadioGroup } from './EmojiRadioGroup';
export { FirstLetterIcon } from './FirstLetterIcon';
export { Icon } from './Icon';
export { IconButton } from './IconButton';
export { IconUpload } from './IconUpload';
export { Link } from './Link';
export { Modal } from './Modal';
export { Pagination } from './Pagination';
export { Paper } from './Paper';
export { SelectableEmoji } from './SelectableEmoji';
export { SideBarListItem } from './SideBarListItem';
export { SpeedDial } from './SpeedDial';
export { Stepper } from './Stepper';
export { TextField } from './TextField';
export { TimeLineItem } from './TimeLineItem';
export { TrimImageModal } from './TrimImageModal';
export { Typography } from './Typography';
