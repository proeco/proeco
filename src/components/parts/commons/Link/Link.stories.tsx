import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Box } from '@mui/system';
import { Link } from './Link';
import { Typography } from '~/components/parts/commons/Typography';

export default {
  title: 'parts/commons/atoms/Link',
  component: Link,
} as ComponentMeta<typeof Link>;

const Template: ComponentStory<typeof Link> = ({ ...rest }) => {
  return (
    <Box>
      <Link {...rest}>
        <Typography>Linkのテキスト</Typography>
      </Link>
    </Box>
  );
};

export const Default = Template.bind({});
Default.args = {
  href: '/',
};
