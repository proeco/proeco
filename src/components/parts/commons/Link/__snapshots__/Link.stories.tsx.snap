// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots parts/commons/Link Default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-e3mlp-MuiTypography-root-MuiLink-root"
          href="/"
          rel="noopener noreferrer"
          target="_self"
        >
          <p
            bold="0"
            class="MuiTypography-root MuiTypography-body1 css-61ooy4-MuiTypography-root"
          >
            Linkのテキスト
          </p>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-e3mlp-MuiTypography-root-MuiLink-root"
        href="/"
        rel="noopener noreferrer"
        target="_self"
      >
        <p
          bold="0"
          class="MuiTypography-root MuiTypography-body1 css-61ooy4-MuiTypography-root"
        >
          Linkのテキスト
        </p>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
